dht/token.hh

```diff
@@ -42,7 +42,9 @@ public:
     kind _kind;
     int64_t _data;
 
-    token() : _kind(kind::before_all_keys) {
+    token()
+        : _kind(kind::before_all_keys)
+        , _data(0) {
```

Comment:
Should we initialize it to min()? then long_token() doesn't need to check the kind.

Reply:
That should work since we don't rely on a particular value in data for min/max tokens.

Reply:
@avikivity actually I take my words back, as the existing code compares tokens against the minimum of maximum tokens like the following:
```diff
diff --git a/db/size_estimates_virtual_reader.cc b/db/size_estimates_virtual_reader.cc
index a8979f66d1..b9fd4cecf4 100644
--- a/db/size_estimates_virtual_reader.cc
+++ b/db/size_estimates_virtual_reader.cc
@@ -195,10 +195,10 @@ static future<std::vector<token_range>> get_local_ranges(replica::database& db,
         // All queries will be on that table, where all entries are text and there's no notion of
         // token ranges form the CQL point of view.
         auto left_inf = boost::find_if(ranges, [] (auto&& r) {
-            return r.end() && (!r.start() || r.start()->value() == dht::minimum_token());
+            return r.end() && (!r.start() || r.start()->value().is_minimum());
         });
         auto right_inf = boost::find_if(ranges, [] (auto&& r) {
-            return r.start() && (!r.end() || r.end()->value() == dht::maximum_token());
+            return r.start() && (!r.end() || r.end()->value().is_maximum());
         });
         if (left_inf != right_inf && left_inf != ranges.end() && right_inf != ranges.end()) {
             local_ranges.push_back(token_range{to_bytes(right_inf->start()), to_bytes(left_inf->end())});
```

And currently min/max tokens are using 0 in the _data member.

so if using `min()` for the in-memory token, we need to serialize the data value as 0
and convert it to std::min upon deserialization.
I'm not sure it's worth it...